### DISCUSSION TITLE
Using non-UD Arabic data

### DISCUSSION THREAD

**Discussion by muhammed-abuodeh** [explosion/spaCy#9333] [Language Support]
I'm trying to convert Conll-U data, though the tags are not in UD. I went over the master thread and didn't find information on how to use the tag property when converting.

Where can I find this information?

Thanks in advance!

**Comment by polm** [comment]
There is no special option for this. I've never heard anyone ask for it before...

If you look at the [conllu format](https://universaldependencies.org/format.html), there should be a UD POS field and a non-UD POS field just after it. Do you have a filler value in your UD POS field or something? If it's a valid value (like X I think?) then it should be able to just read it in. 

I would try just converting your data, and if you have an error I think you can modify the source to make the necessary changes - look in [this file](https://github.com/explosion/spaCy/blob/master/spacy/training/converters/conllu_to_docs.py).

**Reply by muhammed-abuodeh** [reply]
Thanks for your response.

I see what you mean; the UPOS is for UD and XPOS is the format I assume you're referring to.

I asked because I got the following error when UPOS wasn't in UD:
`ValueError: [E1021] pos value "VRB" is not a valid Universal Dependencies tag. Non-UD tags should use the tag property.`

that's why I asked about the tag property. I'll look into the UPOS and XPOS, and I'll check out the code you shared.

Thanks!

**Reply by polm** [reply]
OK, thanks for clarifying. It does sound like you'll need to either modify your data or the converter. I've never heard of a conllu file that puts non-UPOS in the UPOS field, the XPOS field exists so you don't have to do that.
